File: Ct/Pattern Printing - WIPR 001/solution.c
#include <stdio.h>
void printPattern(int N)
{
    int x=1;
    for(int i=1;i<=N;i++){
        if(i%2){
            for(int j=0;j<i;j++){
                printf("%d ",x++);
            }
        }
        else{
            for(int j=0;j<i;j++) x++;
            for(int j=x-1;j>=x-i;j--) printf("%d ",j);
        }
        printf("\n");
    }
}
int main(){
    int n;
    scanf("%d",&n);
    printPattern(n);
}

File: Ct/Pattern Printing - WIPR 001/solution.java
private static void printPattern(int N) {
    int t=1;
    for(int i=1;i<=N;i++){
        if(i%2==0){
            t+=i-1;
            for(int j=0;j<i;j++)
            System.out.print((t--)+" ");
            t+=i+1;
        }
        else{
            for(int j=0;j<i;j++)
            System.out.print((t++)+" ");
        }
        System.out.println();
    }
}